tenpy/networks/mps.py

```diff
@@ -3114,6 +3199,64 @@ def charge_variance(self, bond=0):
         charges, ps = self.probability_per_charge(bond)
         return np.sum(ps[:, np.newaxis] * (charges - charges_mean[np.newaxis, :])**2, axis=0)
 
+    @staticmethod
+    def get_charge_tree_for_given_charge_sector(sites: list, charge_sector: tuple):
+        """Construct the charge-tree for a given charge sector.
+
+        This is a tree of possible charges for each site s.t. the MPS lies in the given `charge_sector`.
+
+        Parameters
+        ----------
+        sites : list of :class:`~tenpy.networks.site.Site`
+            The sites defining the local Hilbert space. The sites should conserve *some* charge,
+            otherwise projecting onto a charge sector is meaningless.
+        charge_sector : tuple of int
+
+        Returns
+        -------
+        charge_tree : list
+        """
+        # TODO: should we accept an integer as charge sector? charge_sector = tuple([charge_sector]) ?
```

Comment:
no, the usual definition is a tuple/list with as many entries as there are charges. That's common througout TeNPy.